tests/test_weaviate_adaptor.py

```diff
@@ -87,10 +88,13 @@ def dataset_fixture(N, dim) -> Dataset:
 def test_constructor(dataset, N):
     """Test the constructor of the Weaviate vector database."""
     adaptor = Weaviate(dataset, Metric.COSINE, "flat")
-    collection = adaptor.client.collections.use(adaptor.collection_name)
+    try:
+        collection = adaptor.client.collections.use(adaptor.collection_name)
 
-    response = collection.aggregate.over_all(total_count=True)
-    assert response.total_count == N
+        response = collection.aggregate.over_all(total_count=True)
+        assert response.total_count == N
+    finally:
+        adaptor.close()
```

Comment:
We shouldn't need this since I overrode the `__del__` method which will automatically call `adaptor.close()` on GC.